### PR TITLE
[WIP] Refactor engine, remove execution graphs

### PR DESCRIPTION
**Pull request by cjolivier01**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by cjolivier01** [comment]
Reverting commits to find where CI was broken

File: src/operator/nn/lrn.cc [hunk 1]
```diff
@@ -181,14 +181,6 @@ number of kernels in the layer.
...
 .set_attr<nnvm::FInferType>("FInferType", LRNType)
 .set_attr<FInferStorageType>("FInferStorageType", LRNForwardInferStorageType)
-.set_attr<nnvm::FListInputNames>("FListInputNames",
-    [](const NodeAttrs& attrs) {
-  return std::vector<std::string>{"data"};
-})
-.set_attr<nnvm::FListInputNames>("FListOutputNames",
-    [](const NodeAttrs& attrs) {
-  return std::vector<std::string>{"output", "tmp_norm"};
-})
```
Comment (zheng-da) [inline src/operator/nn/lrn.cc]: Why are these attributes removed? They are just added to address a broken feature.

File: src/operator/nn/fully_connected.cc [hunk 2]
```diff
@@ -267,7 +267,7 @@ This could be used for model inference with `row_sparse` weights trained with `S
     return std::vector<std::string>{"data", "weight"};
   }
 })
-.set_attr<nnvm::FListOutputNames>("FListOutputNames",
```
Comment (zheng-da) [inline src/operator/nn/fully_connected.cc]: I think @reminisce has a PR to fix this.

File: src/executor/graph_executor.cc [hunk 3]
```diff
@@ -1360,6 +1359,8 @@ void GraphExecutor::InitOpSegs() {
   if (prefer_bulk_exec_inference && !is_training) {
     this->BulkInferenceOpSegs();
   }
+
+  return;
```
Comment (zheng-da) [inline src/executor/graph_executor.cc]: why do we need a return at the end of a function?

File: src/engine/threaded_engine.cc [hunk 4]
```diff
@@ -326,17 +326,18 @@ void ThreadedEngine::PushSync(SyncFn exec_fn, Context exec_ctx,
...
         exec_fn(ctx);
         on_complete();
       }, exec_ctx, const_vars, mutable_vars, prop, priority, opr_name);
     return;
   }
 
-  const BulkStatus& bulk_status = *BulkStatusStore::Get();
   if (bulk_status.count && exec_ctx != bulk_status.ctx) BulkFlush();
   BulkAppend(exec_fn, exec_ctx, const_vars, mutable_vars);
+  return;
```
Comment (zheng-da) [inline src/engine/threaded_engine.cc]: the same here. Is this a new practice?

File: src/operator/nn/lrn.cc (hunk 1 above)
Comment (cjolivier01) [reply src/operator/nn/lrn.cc]: I didn't think they visually went with the flow of the rest of the code. Plus there's a lot of confusing letters and stuff. Shouldn't this be written in Perl instead?

File: src/executor/graph_executor.cc (hunk 3 above)
Comment (cjolivier01) [reply src/executor/graph_executor.cc]: It looks cool

**Comment by cjolivier01** [comment]
ok, silly people -- this is just a PR reverting commits to find where the CI went bad
No need to review lol

File: src/engine/threaded_engine.cc (hunk 4 above)
Comment (cjolivier01) [reply src/engine/threaded_engine.cc]: superfluous returns are back in style :D

File: src/operator/nn/lrn.cc (hunk 1 above)
Comment (zheng-da) [reply src/operator/nn/lrn.cc]: please check out this: https://github.com/apache/incubator-mxnet/issues/10282
It turns out the output names are used by other frameworks.

**Comment by zheng-da** [comment]
Ah, I see :)

**Comment by marcoabreu** [comment]
"Refactor engine, remove execution graphs" haha lets just refactor the engine to just return success immediately! It's way less resource consuming if we just skip the entire graph execution anyways :P - and we won't have any hangs! You're a genius!

**Pull request closed without merging**